src/Nowruz/pages/search/useSearch.tsx

```diff
@@ -1,169 +1,176 @@
+import { COUNTRIES_DICT } from 'src/constants/COUNTRIES';
+import { Job, JobsRes, Organization, OrganizationsRes, User, UsersRes } from 'src/core/api';
+import { search as searchReq } from 'src/core/api/site/site.api';
+import { isTouchDevice } from 'src/core/device-type-detector';
+import { removeValuesFromObject } from 'src/core/utils';
+import { JobListingCard } from 'src/Nowruz/modules/Jobs/components/JobListingCard';
+import { SearchResultProfile } from 'src/Nowruz/modules/Search/components/searchResultProfile';
+
+export type FilterReq = {
+  causes_tags?: Array<string>;
+  skills?: Array<string>;
+  country?: Array<keyof typeof COUNTRIES_DICT>;
+  city?: Array<string>;
+  remote_preference?: string;
+  job_category_id: string;
+  project_length?: Array<string>;
+  experience_level: Array<number>;
+  payment_type?: string | number;
+  location?: { value: number; label: string; countryCode: string };
+};
+
+export const useSearch = () => {
+  const { searchData: data } = useLoaderData() as { searchData: JobsRes | UsersRes | OrganizationsRes };
+  const [searchParams, setSearchParams] = useSearchParams();
+  const type = searchParams.get('type');
+  const q = searchParams.get('q');
+
+  const PER_PAGE = 10;
+  const isMobile = isTouchDevice();
+  const [searchResult, setSearchResult] = useState({} as JobsRes | UsersRes | OrganizationsRes);
+  const [page, setPage] = useState(data.page);
+  const [sliderFilterOpen, setSliderFilterOpen] = useState(false);
+  const savedFilter = JSON.parse(localStorage.getItem('filter') || '{}') as FilterReq;
+  const [filter, setFilter] = useState<FilterReq>(savedFilter);
+  const [countryName, setCountryName] = useState<string | undefined>('');
+
+  const navigate = useNavigate();
+  function getCountryName(shortname?: keyof typeof COUNTRIES_DICT | undefined) {
+    if (shortname && COUNTRIES_DICT[shortname]) {
+      return COUNTRIES_DICT[shortname];
+    } else {
+      return shortname;
+    }
+  }
+
+  const filterNeeded = (filter: FilterReq) => {
+    const propertyName = type === 'projects' ? 'causes_tags' : 'social_causes';
+    return {
+      [propertyName]: filter.causes_tags,
+      skills: filter.skills,
+      country: filter.country,
+      city: filter.city,
+      remote_preference: filter.remote_preference,
+      job_category_id: filter.job_category_id,
+      project_length: filter.project_length,
+      experience_level: filter.experience_level,
+      payment_type: filter.payment_type,
+    };
+  };
+
+  const fetchMore = async (page: number) => {
+    const body = {
+      filter: filter ? removeValuesFromObject(filterNeeded(filter), ['', null, undefined]) : {},
+      type,
+    } as any;
+    if (q?.trim()) {
+      Object.assign(body, { q });
+    }
+    const data = await searchReq(body, { limit: 10, page });
+    if (isMobile && page > 1) setSearchResult({ ...searchResult, ...data });
+    else setSearchResult(data);
+  };
+
```

Comment:
I cannot see where this function is called. It seems filters are not passed to search function anymore

Reply:
@marjan-ara 
Elham is not available, so if you think change is necessary, feel free to do it.